test/src/test/java/org/corfudb/integration/LogReplicationIT.java

```diff
@@ -1046,6 +1051,145 @@ private void testSnapshotSyncAndLogEntrySync(int numCyclesToDelayApply, boolean
         verifyPersistedLogEntryMetadata();
     }
 
+     /**
+     * Test Log Entry (delta) Sync for the case where the ACKs are arbitrarily dropped
+     * for a fixed number of times at the Source. This will test that LR is
+     * (i) not impacted by dropped ACKs
+     * (ii) Source resends msgs for which it hasn't received the ACKs
+     * (iii) Sink handles the already seen and processed msgs.
+     **/
+    @Test
+    public void testLogEntrySyncWithAckDrops() throws Exception {
+        setupEnv();
+        // Write data in transaction to t0 and t1
+        Set<String> crossTables = new HashSet<>();
+        crossTables.add(t0);
+        crossTables.add(t1);
+
+        // Writes transactions to t0, t1 and t2 + transactions across 'crossTables'
+        writeCrossTableTransactions(crossTables, true);
+
+        // Start Log Entry Sync
+        expectedAckMessages = NUM_KEYS * WRITE_CYCLES;
```

Comment:
if miss one ack, should this expectedAckMessages -1?

Reply:
No. Active keeps msgs in a queue and resends them until ACKs are received (Msgs are evicted from that queue when ack.timestamp >= msg.timestamp).